db.go

```diff
@@ -336,6 +337,74 @@ func Open(opt Options) (db *DB, err error) {
 	return db, nil
 }
 
+func (db *DB) IngestExternalFiles(files []*os.File) error {
+	tbls, err := db.prepareExternalFiles(files)
+	if err != nil {
+		return err
+	}
+
+	if err := db.checkExternalTables(tbls); err != nil {
+		return err
+	}
+
+	task := &ingestTask{tbls: tbls}
+	task.Add(1)
+	db.ingestCh <- task
+	task.Wait()
+	return task.err
+}
+
+func (db *DB) prepareExternalFiles(files []*os.File) ([]*table.Table, error) {
+	tbls := make([]*table.Table, len(files))
+	for i, fd := range files {
+		id := db.lc.reserveFileID()
+		filename := table.NewFilename(id, db.opt.Dir)
+
+		err := os.Link(fd.Name(), filename)
+		if err != nil {
+			return nil, err
+		}
+
+		fd, err := os.OpenFile(fd.Name(), os.O_RDWR, 0666)
+		if err != nil {
+			return nil, err
+		}
+
+		tbl, err := table.OpenTable(fd, db.opt.TableLoadingMode)
+		if err != nil {
+			return nil, err
+		}
+
+		tbls[i] = tbl
+	}
+
+	return tbls, syncDir(db.lc.kv.opt.Dir)
+}
+
+func (db *DB) checkExternalTables(tbls []*table.Table) error {
+	for i, t := range tbls {
+		for j, o := range tbls {
+			if i == j {
+				continue
+			}
+
+			cmp := bytes.Compare(t.Smallest(), o.Smallest())
```

Comment:
We can collect the smallest and biggest keys of the tables and call `sort.SliceIsSorted`